remove-support-for-sphinx-1.x.rst

```diff
@@ -0,0 +1,72 @@
+.. post:: Jun 5, 2023
+   :tags: builders, deprecation
+   :author: Manuel
+   :location: BCN
+   :category: Changelog
+
+Remove support for Sphinx 1.x
```

Comment:
I think this title doesn't communicate very clearly what's happening. Based on the body, users can still use Sphinx 1.x if they want as long as they pin it. I'd frame this more as changing the default version of Sphinx.